RocketChat/Rocket.Chat
# feat: added modal confirmation before pinning message

Proposed changes (including videos or screenshots)

AFTER THE FIX
https://github.com/RocketChat/Rocket.Chat/assets/112304873/2e74a44f-e168-48af-817d-c7b945d1f5b0

REASON BEHIND CHANGES
Made the same changes as done by DISCORD to tackle this issue.

Issue(s)
This should close issues :-
closes https://github.com/RocketChat/Rocket.Chat/issues/30777

Steps to test or reproduce

1. Go to Teams/Channel
2. Click on any of the teams/Channels where you are participant.
3. Try to Send a Text to the Team/Channel
4. Try to Edit the Text and See that there is no Cancel Button for Edit Text if the user wants to continue without editing the Text.]
